src/ex6/client/src/components/TodosHeader.js

```diff
@@ -1,60 +1,86 @@
 import React, { useState, useRef, useEffect } from "react";
-import styles from "../style/TodosHeader.module.css";
-import { addTodo, removeAllTodos } from "../todos-client";
 import { MdOutlineAddCircle, MdDeleteSweep } from "react-icons/md";
+import { connect } from "react-redux";
+
+import { addTodoAction } from "../actions/add-todo-action";
+import { removeAllTodoAction } from "../actions/remove-all-todos-action";
+import { unDeleteTodoAction } from "../actions/undelete-todo-action";
+import { filterTodosAction } from "../actions/filter-todos-action";
+import { Hide, Show, Mirror } from "monday-ui-react-core/dist/allIcons";
+
+import styles from "../style/TodosHeader.module.css";
 import {
   DEFAULT_PLACEHOLDER_TEXT,
   UNVALID_PLACEHOLDER_TEXT,
 } from "../constants";
-export default function TodosHeader({ onItemAdd }) {
+
+function TodosHeader({
+  addTodoAction,
+  removeAllTodoAction,
+  filterTodosAction,
+  unDeleteTodoAction,
+  todos,
+}) {
   const [inputText, setInputText] = useState("");
   const [unvalidInput, setUnvalidInput] = useState(false);
+  const [hideIcon, setHideIcon] = useState(true);
   const inputRef = useRef();
 
   useEffect(() => {
     inputRef.current.focus();
   }, [unvalidInput]);
 
-  const addNewTodo = async () => {
+  const addNewTodo = () => {
     if (inputText.length === 0) {
       setUnvalidInput(true);
       return;
     }
-    await addTodo(inputText);
-    onItemAdd();
+    addTodoAction(inputText);
     setUnvalidInput(false);
     setInputText("");
+    filterTodosAction("SEARCH", todos, "");
   };
 
-  const deleteAllTodos = async () => {
-    await removeAllTodos();
-    onItemAdd();
+  const deleteAllTodos = () => {
+    removeAllTodoAction();
   };
 
-  const enterPressed = async (event) => {
+  const enterPressed = (event) => {
     if (event.key === "Enter") {
-      await addNewTodo();
+      addNewTodo();
     }
   };
 
+  const filterIconPressed = () => {
+    setHideIcon(!hideIcon);
+    filterTodosAction("UNDONE", todos);
+  };
+
+  const onChangeEvent = (e) => {
+    setInputText(e.target.value);
+    filterTodosAction("SEARCH", todos, e.target.value);
+  };
+
   let placeHolderText = unvalidInput
     ? UNVALID_PLACEHOLDER_TEXT
     : DEFAULT_PLACEHOLDER_TEXT;
 
   return (
     <div>
-      <h1 className={styles.title}>My Todo App</h1>
       <div className={styles.addTask}>
+        <button className={styles.undoneBtn} onClick={filterIconPressed}>
+          {hideIcon ? <Show /> : <Hide />}
+        </button>
         <input
           ref={inputRef}
-          value={inputText} //defaultValue is not working
+          value={inputText}
           placeholder={placeHolderText}
           className={`${styles.inputHeader} ${
             unvalidInput ? styles.redPlaceholder : ""
           } `}
           type="text"
           required
-          onChange={(e) => setInputText(e.target.value)}
+          onChange={(e) => onChangeEvent(e)}
```

Comment:
```suggestion
          onChange={onChangeEvent}
```
no need to create a new ref of function on each render